src/main/java/com/joss/conductor/mobile/ConductorConfig.java

```diff
@@ -502,6 +505,28 @@ public String getFullAppPath() {
         }
 
         Path path = Paths.get(appFile);
+        try {
+            if (platformName == Platform.ANDROID) {
+                    File dir = new File(appFile);
+                    File[] files = dir.listFiles((dir1, name) -> name.toLowerCase().endsWith(".apk"));
+                    if (files != null) {
+                        File apkName = files[0];
+                        Logger.warn("Found an APK at relative path" + apkName);
+                        return Paths.get(System.getProperty("user.dir"), apkName.toString()).normalize().toString();
+                    }
+            } else {
+                File dir = new File(appFile);
+                File[] files = dir.listFiles((dir1, name) -> name.toLowerCase().endsWith(".ipa"));
+                if (files != null) {
+                    File apkName = files[0];
+                    Logger.warn("Found an IPA at relative path" + apkName);
+                    return Paths.get(System.getProperty("user.dir"), apkName.toString()).normalize().toString();
```

Comment:
There's a lot of duplication between these two blobs. Better might have been:
```
string fileExtension = platformName == Platform.Android ? "apk" : "ipa";
File dir = new File(appFile);
File[] files = dir.listFiles((dir1, name) -> name.toLowerCase().endsWith(fileExtension));
// etc.
```

Alternately, it may make more sense to let this function instead use glob patterns itself. So instead of assuming that appFile is a directory, assume it's either a file that exists or a "glob" and take the first match on the glob. Not sure if this will help but looks promising: https://javapapers.com/java/glob-with-java-nio/

Reply:
Gonna give this a shot tonight when I get time. Thank you @fuzzybinary !

Reply:
Awesome suggestion, I may have to change from the elvis operator to include logic for also finding an `.app` file for simulators along with an `.ipa` for devices. Right now the logic is defaulting to only looking for an `.apk` or an `.app`